interview-questions.md

```diff
@@ -9,59 +9,59 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
+  Your answer: The required lifecycle method is Render(). I do not remember exactly why but I remember hearing this lol.
 
-  Researched answer:
+  Researched answer: Render is the only lifecycle method required, and it sends the output HTML to the DOM. When a component gets updated, it is the Render method that is called to make sure the DOM gets the new HTML changes. Render does not interact with the browser or modify the component state, but it returns either: a react element, an array or fragment of elements, portals, strings, numbers, bools, or null.
 
 
 
 4. What is JSX? What is one notable difference between HTML and JSX?
 
-  Your answer:
+  Your answer: JSX is a type of syntax, used in React, to return multiple "things" at once. Since a React component can only return one "thing" at a time, by wrapping multiple "things" in a JSX tag, you can return them all in between the JSX tags. It almost seems like a built in method, to push the elements within the JSX tag into an array, and just return the single array (my random thought lol). The difference between HTML and JSX, is that you don't write HTML in JavaScript files, you use JSX.
 
-  Researched answer:
+  Researched answer: I also found that JSX is JavaScript and XML mixed together, to allow for the rendering of HTML to the DOM. A technology called "Webpack" converts the JSX into HTML and JavaScript for the browser to read; therefor, HTML is a language understood by the browser whereas JSX needs to be translated before anything actually happens on the webpage. You can also write JavaScript directly into JSX by placing it in: {}, opposed to HTML where you need to use a script tag. HTML attributes must also be written in camelCase, when using JSX (ex. onclick vs onClick).
 
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
+  Your answer: Yarn is a command used to create and open a react app. It imports the needed dependencies to create a "shell" react app. When yarn is ran, all files saved on the React application are ran, and any changes made since the previous save are changed in the application.
 
-  Researched answer:
+  Researched answer: I found that yarn is a package manager that was created by FB as an alternative to the npm package manager. It allows you to still access npm dependencies but is more efficient.
```

Comment:
Yes, yarn will connect dependencies. The yarn.lock, package.json, and node_nodules will be updated.